core/server/server.js

```diff
@@ -494,65 +487,65 @@ class Server {
 
     log.log(`Server is starting up: ${this.baseUrl}`)
 
-    const camp = (this.camp = Camp.create({
-      documentRoot: this.config.public.documentRoot,
-      port,
-      hostname,
-      secure,
-      staticMaxAge: 300,
-      cert,
-      key,
-    }))
+    const app = (this.app = express())
 
     if (requireCloudflare) {
       this.requireCloudflare()
     }
 
     const { githubConstellation, metricInstance } = this
-    await githubConstellation.initialize(camp)
+    await githubConstellation.initialize(app)
     if (metricInstance) {
       if (this.config.public.metrics.prometheus.endpointEnabled) {
-        metricInstance.registerMetricsEndpoint(camp)
+        metricInstance.registerMetricsEndpoint(app)
       }
       if (this.influxMetrics) {
         this.influxMetrics.startPushingMetrics()
       }
     }
 
     const { apiProvider: githubApiProvider } = this.githubConstellation
-    setRoutes(allowedOrigin, githubApiProvider, camp)
+    setSuggestRoutes(allowedOrigin, githubApiProvider, app)
 
     // https://github.com/badges/shields/issues/3273
-    camp.handle((req, res, next) => {
+    app.use((req, res, next) => {
       res.setHeader('Access-Control-Allow-Origin', '*')
       next()
     })
 
     this.registerErrorHandlers()
     this.registerRedirects()
+    app.use(
+      express.static(this.config.public.documentRoot, {
+        // Since express's `maxAge` parameter sets `Cache-Control: public`, set
+        // the headers manually insetad.
+        cacheControl: false,
+        setHeaders: res =>
+          res.setHeader('Cache-Control', 'max-age=300, s-maxage=300'),
+      })
+    )
     await this.registerServices()
+    if (registerExtras) {
+      registerExtras(app)
+    }
+    this.registerNotFoundHandlers()
 
-    camp.timeout = this.config.public.requestTimeoutSeconds * 1000
-    if (this.config.public.requestTimeoutSeconds > 0) {
```

Comment:
Replace with Express' built-in timeout support. It doesn't return a 408; I think it just hangs up the socket.